From stable-bounces@linux.kernel.org Mon Oct 16 05:38:07 2006
Date: Mon, 16 Oct 2006 14:37:31 +0200
Message-ID: <s5hpscsa12s.wl%tiwai@suse.de>
From: Takashi Iwai <tiwai@suse.de>
To: stable@kernel.org
MIME-Version: 1.0 (generated by SEMI 1.14.6 - "Maruoka")
Subject: ALSA: powermac - Fix Oops when conflicting with aoa driver
Content-Type: text/plain; charset="us-ascii"

From: Takashi Iwai <tiwai@suse.de>

[PATCH] ALSA: powermac - Fix Oops when conflicting with aoa driver

Fixed Oops when conflictin with aoa driver due to lack of
i2c initialization.

Signed-off-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>

---
 sound/ppc/keywest.c |    3 +++
 1 file changed, 3 insertions(+)

--- linux-2.6.18.1.orig/sound/ppc/keywest.c
+++ linux-2.6.18.1/sound/ppc/keywest.c
@@ -117,6 +117,9 @@ int __init snd_pmac_tumbler_post_init(vo
 {
 	int err;
 	
+	if (!keywest_ctx || !keywest_ctx->client)
+		return -ENXIO;
+
 	if ((err = keywest_ctx->init_client(keywest_ctx)) < 0) {
 		snd_printk(KERN_ERR "tumbler: %i :cannot initialize the MCS\n", err);
 		return err;
